beacon-chain/state/getters.go

```diff
@@ -413,6 +413,32 @@ func (b *BeaconState) stateRoots() [][]byte {
 	return b.safeCopy2DByteSlice(b.state.StateRoots)
 }
 
+// StateRootAtIndex retrieves a specific state root based on an
+// input index value.
+func (b *BeaconState) StateRootAtIndex(idx uint64) ([]byte, error) {
+	if !b.HasInnerState() {
+		return nil, ErrNilInnerState
+	}
+	if b.state.StateRoots == nil {
+		return nil, nil
+	}
+
+	b.lock.RLock()
```

Comment:
i think the lock should be before `b.state.StateRoots == nil`? It's a read there

Reply:
True, although this is a general pattern in all our getters which in hindsight looks incorrect. I can fix it in another pr.